packages/x-date-pickers/src/DesktopTimePicker/tests/DesktopTimePicker.test.tsx

```diff
@@ -153,9 +151,8 @@ describe('<DesktopTimePicker />', () => {
 
       await user.click(screen.getByRole('option', { name: 'PM' }));
       expect(onChange.callCount).to.equal(4);
-      expect(onAccept.callCount).to.equal(1);
-      expect(onAccept.lastCall.args[0]).toEqualDateTime(new Date(2018, 0, 1, 14, 25));
-      expect(onClose.callCount).to.equal(1);
+      expect(onAccept.callCount).to.equal(0); // onAccept will only called when onCloseOnSelect is true (false by default for pickers with time view)
+      expect(onClose.callCount).to.equal(0); // onClose will only called when onCloseOnSelect is true (false by default for pickers with time view)
```

Comment:
Could we either replace `onAccept` assertion with `onChange` and remove the other ones?
Or add a `closeOnSelect` prop and keep the test as is? 🤔 
```suggestion
      expect(onChange.lastCall.args[0]).toEqualDateTime(new Date(2018, 0, 1, 14, 25));
```